examples/dynamic-bridge-app/linux/include/data-model/DataModel.h

```diff
@@ -173,24 +192,20 @@ CHIP_ERROR Decode(const ConcreteDataAttributePath &, AttributeValueDecoder & aDe
  * @brief
  * Lists that are string-like should be decoded as char/byte spans.
  */
-template <
-    typename X,
-    std::enable_if_t<IsList<std::decay_t<X>>::value && sizeof(std::decay_t<typename X::pointer>) == sizeof(char), bool> = true>
-CHIP_ERROR Decode(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder, X & x)
+template <typename X>
+CHIP_ERROR Decode(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder, StringContainer<X> & x)
 {
     Span<std::decay_t<typename X::pointer>> span;
     CHIP_ERROR err = aDecoder.Decode(span);
     if (err == CHIP_NO_ERROR)
     {
-        x = X(span.data(), span.size());
+        x.assign(span.data(), span.data() + span.size());
     }
     return err;
 }
 
-template <
-    typename X,
-    std::enable_if_t<IsList<std::decay_t<X>>::value && (sizeof(std::decay_t<typename X::pointer>) > sizeof(char)), bool> = true>
-CHIP_ERROR Decode(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder, X & x)
+template <typename X>
+CHIP_ERROR Decode(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder, ListContainer<X> & x)
```

Comment:
The ReplaceAll behavior in this function does not look right to me.  Shouldn't it end up decoding multiple entries from the list as needed?  Or none if there are none?  Right now it does neither one correctly: it seems to assume that ReplaceAll will come with a single entry...

Reply:
No, it appears to come from here: https://github.com/project-chip/connectedhomeip/blob/8eb1865531ad65f530983c1ddc4078a9fc94f4d8/src/app/WriteHandler.cpp#L312

So it will only be set if there is no other list operation specified on the write, plus there is no expectation of decoding multiple entries for other types of writes (there appears to be metadata encoded between entries anyway). I'm not sure how a non-nullable empty list would be represented here as write is called unconditionally

Reply:
> So it will only be set if there is no other list operation specified on the write

Right, which is what happens with non-chunked writes.

>  plus there is no expectation of decoding multiple entries for other types of writes

Yes, because those can only have one entry, not a list.

> I'm not sure how a non-nullable empty list would be represented here as write is called unconditionally

For ReplaceAll you basically need to decode into a DecodableList, then iterate it.